helm-charts/basehub/values.yaml

```diff
@@ -1447,17 +1455,32 @@ jupyterhub:
                   timeout=spawner.k8s_api_request_retry_timeout
               )
           c.Spawner.pre_spawn_hook = ensure_db_pvc
-      05-profile-groups: |
-        # Re-assignes c.KubeSpawner.profile_list to a callable that filters the
-        # initial configuration of profile_list based on the user's github
-        # org/team membership as declared via "allowed_groups" read from
-        # profile_list profiles.
+      05-auth-state-groups-key: |
+        # Pass the user's GitHub teams from auth_state as groups to the authenticator
+        async def custom_auth_state_groups_key(auth_state):
+          groups_list = []
+          if auth_state is None:
+            print("Auth state is None")
+            return groups_list
+          else:
+            for team in auth_state["teams"]:
+              if f'{team["organization"]["login"]}:{team["slug"]}' not in c.GitHubOAuthenticator.allowed_organizations:
```

Comment:
We should unconditionally put all teams into groups, as we may change what's in `c.GitHubOAuthenticator.allowed_organizations` later on. And if we do it conditionally, we'll need to force people to re-log in. This does possibly create extra entries, but I think that's ok.

Reply:
I thought about this and decided against doing this unconditionally.

Users grant `read:org` access to populate GH teams in the `auth_state` that could be completely unrelated to the hub that will populate the JH groups table in the database and therefore pass to other places like the `jupyterhub_groups_exporter`.

Therefore to keep authorisation up to date with `c.GitHubOAuthenticator.allowed_organizations` we should rely on users to log in again, which I will write up separately.

Reply:
Here's my login/logout strategy writeup: https://github.com/2i2c-org/infrastructure/pull/6032#issuecomment-2944180023.

> And if we do it conditionally, we'll need to force people to re-log in.

I think having it conditional to protect private GH team memberships is okay for now: information from `c.GitHubOAuthenticator.auth_state_groups_key` will only be used to populate JupyterHub groups. JupyterHub groups is currently only used for the `jupyterhub_groups_exporter`, which is not on the critical path for hub functionality such as profile list allowed groups, server launches.

Therefore the extra config in `05-profile-groups` should stay the same and depend on team memberships from the GH auth state and not JH group memberships for now.